src/sinks/kafka.rs

```diff
@@ -392,6 +405,28 @@ impl Sink<Event> for KafkaSink {
     }
 }
 
+fn get_headers(event: &Event, headers_key: &str) -> Option<OwnedHeaders> {
+    if let Event::Log(log) = event {
+        if let Some(headers) = log.get(headers_key) {
+            match headers {
+                Value::Map(headers_map) => {
+                    let mut owned_headers = OwnedHeaders::new_with_capacity(headers_map.len());
+                    for (key, value) in headers_map {
+                        if let Value::Bytes(value_bytes) = value {
+                            owned_headers = owned_headers.add(key, value_bytes.as_ref());
+                        } else {
+                            warn!("Header value for key {} must be a UTF-8 encoded String (Bytes). Ignoring.", key);
```

Comment:
This message isn't quite correct, since neither `Bytes` nor Kafka header values are required to be UTF-8. Both are fine to accept arbitrary byte strings.

Also, the smallest of nitpicks, but we tend to prefer using `tracing`'s [structured fields](https://docs.rs/tracing/0.1.28/tracing/index.html#recording-fields) rather than string interpolation for log messages. I mention that for context more than anything else, since it's not a hard rule by any stretch.

Reply:
I'm getting rid of these messages anyway (switching it to an internal event), but thanks! I agree with you, the structured fields seem nicer.